src/azure-cli-core/azure/cli/core/decorators.py

```diff
@@ -0,0 +1,93 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+Utility decorators
+
+The module variable is_diagnostics_mode can be set to a boolean or callable. It is used to determine
+if an exception should be re-raised when the suppression decorator is requested to raise in
+diagnostics mode.
+"""
+
+# NOTE: Do not import modules other than the ones of Python Standard Library.
```

Comment:
Hope to keep this module clean, since it is referenced in `telemetry_core.py`. The later module will be executed in separate process after az process is terminated to upload traces. I would hope the second process not to import `azure cli` modules.